```
======================================================================
ERROR: setUpClass (corehq.apps.app_manager.tests.test_report_config.ReportFiltersSuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/app_manager/tests/test_report_config.py", line 136, in setUpClass
    fixture, = report_fixture_generator(cls.user, '2.0')
  File "/mnt/corehq/apps/app_manager/fixtures/mobile_ucr.py", line 42, in __call__
    reports_elem.append(self._report_config_to_fixture(report_config, user))
  File "/mnt/corehq/apps/app_manager/fixtures/mobile_ucr.py", line 60, in _report_config_to_fixture
    report = get_report_config(report_config.report_id, user.domain)[0]
  File "/mnt/corehq/apps/userreports/models.py", line 645, in get_report_config
    config = get_report_configs([config_id], domain)[0]
  File "/mnt/corehq/apps/userreports/models.py", line 629, in get_report_configs
    get_docs(ReportConfiguration.get_db(), dynamic_report_config_ids)
  File "submodules/dimagi-utils-src/dimagi/utils/couch/bulk.py", line 85, in get_docs
    url = db.uri + '/_all_docs'
  File "/vendor/lib/python2.7/site-packages/mock.py", line 700, in __getattr__
    raise AttributeError("Mock object has no attribute %r" % name)
AttributeError: Mock object has no attribute 'uri'
```

--- a/corehq/apps/app_manager/tests/mocks/mobile_ucr.py
+++ b/corehq/apps/app_manager/tests/mocks/mobile_ucr.py
@@ -1,4 +1,5 @@
 import mock
+from couchdbkit import ResourceNotFound
 
 
 def mock_report_configurations(report_configurations_by_id):
@@ -8,9 +9,15 @@ def mock_report_configurations(report_configurations_by_id):
 
 
 def mock_report_configuration_get(report_configurations_by_id):
+    def get_report_config(_id, domain):
+        try:
+            return (report_configurations_by_id[_id], False)
+        except KeyError:
+            raise ResourceNotFound
     return mock.patch(
-        'corehq.apps.userreports.models.ReportConfiguration.get',
-        classmethod(lambda self, _id: report_configurations_by_id[_id]))
+        'corehq.apps.app_manager.fixtures.mobile_ucr.get_report_config',
+        get_report_config
+    )
 
 
 def mock_report_data(data):

--- a/corehq/apps/app_manager/tests/test_report_fixtures_provider.py
+++ b/corehq/apps/app_manager/tests/test_report_fixtures_provider.py
@@ -3,7 +3,8 @@ from django.test import SimpleTestCase
 from mock import Mock, patch
 from corehq.apps.app_manager.fixtures.mobile_ucr import ReportFixturesProvider
 from corehq.apps.app_manager.models import ReportAppConfig, StaticChoiceListFilter
-from corehq.apps.app_manager.tests import TestXmlMixin, MAKE_REPORT_CONFIG
+from corehq.apps.app_manager.tests import TestXmlMixin, MAKE_REPORT_CONFIG, \
+    mock_report_configuration_get
 
 
 class ReportFixturesProviderTests(SimpleTestCase, TestXmlMixin):
@@ -30,10 +31,9 @@ class ReportFixturesProviderTests(SimpleTestCase, TestXmlMixin):
         )
         user = Mock()
 
-        with patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportConfiguration') as report_config_patch, \
-                patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportFactory') as report_factory_patch:
+        with mock_report_configuration_get({report_id: MAKE_REPORT_CONFIG('test_domain', report_id)}), \
+            patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportFactory') as report_factory_patch:
 
-            report_config_patch.get.return_value = MAKE_REPORT_CONFIG('test_domain', report_id)
             report_factory_patch.from_spec.return_value = self.get_data_source_mock()
             report = provider._report_config_to_fixture(report_app_config, user)
             self.assertEqual(

--- a/corehq/apps/app_manager/tests/test_report_fixtures_provider.py
+++ b/corehq/apps/app_manager/tests/test_report_fixtures_provider.py
@@ -32,7 +32,7 @@ class ReportFixturesProviderTests(SimpleTestCase, TestXmlMixin):
         user = Mock()
 
         with mock_report_configuration_get({report_id: MAKE_REPORT_CONFIG('test_domain', report_id)}), \
-            patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportFactory') as report_factory_patch:
+                patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportFactory') as report_factory_patch:
 
             report_factory_patch.from_spec.return_value = self.get_data_source_mock()
             report = provider._report_config_to_fixture(report_app_config, user)

